doc/databinding/@property.md

```diff
@@ -66,35 +109,149 @@ Example for a type guard more strict than the compiler:
   }
 ```
 
-### config.type
+By using multiple `@property` decorator on the same property you can give multiple type guards. They are executed in the given order.
 
-**This option is only useful in JavaScript/JSX files.**
 
-Where `type` is a constructor function.
+### config.default
 
-When providing this option the property will check that every assigned value (except `null`) is an instance of the given constructor function, such as `Date`. Primitives (`number`, `string`, `boolean`) are represented by the constructors of their [boxed/wrapped values](https://developer.mozilla.org/en-US/docs/Glossary/Primitive#Primitive_wrapper_objects_in_JavaScript).
+Where `default` can be any value except `undefined`, which turns the feature off.
 
-JavaScript Example:
+Defines the initial value of this property. It will also be used as the fallback value if the property is [not nullable](#confignullable). If a [converter](#configconvert) is defined it will be applied to the default value as well.
+
+Using `default` differs from setting the property with the declaration, which changes the value only *after* `super` has been called. Thus it allows setting the property in a super-constructor:
 
 ```ts
-class Foo {
+@component
+class MyComponent extends Composite {
 
-  /** @type {Date} */
-  @property({type: Date})
-  myDate = new Date();
+  // Bad:
+  // @property num: number = 0;
 
-  /** @type {string} */
-  @property({type: String})
-  myText = 'foo';
+  // Good:
+  @property({default: 0})
+  num: number;
+
+  constructor(properties: Properties<MyComponent>) {
+    super(properties);
+    this.append(/*...*/);
+  }
 
 }
+
+console.log(new MyComponent({}).num); // "0"
+console.log(new MyComponent({num: 1}).num); // "1"
 ```
 
-This is the exact equivalent of the following TypeScript code:
+Alternatively you can use `this.set(properties);` instead of `super(properties);`. In that case it doesn't matter how the property is initialized.
+
+### config.nullable
+
+Where `nullable` is a boolean. The default is `true`.
+
+Dictates whether the property can be set to `null` or `undefined`. If set to `false` the initial value may be `null`/`undefined` (assuming no [default](#configdefault) value is defined), but it can not change to have these values later.
+
+How this is achieved depends on the "[default](#configdefault)" and "[convert](#configconvert)" options. Here is what happens when a non-nullable property is set to `null` or `undefined`:
+
+Target type | Option "default" | Option "convert"      | Result
+------------|------------------|-----------------------|-----------------
+any type    | set              |  any value            | value reset to default
+any type    | not set          | `'off'`               | Exception thrown
+primitive   | not set          | `'auto'` or function  | Value converted
```

Comment:
It's not, it's the natural behavior resulting from an converter beeing applied to any default value, in this case `undefined`. There isn't even any specific code to enable this behavior.

Reply:
As a matter of fact I'm think about changing the behavior so that this behavior extends to the initial state of the property, it would be *more* consistent and remove an "if" statement from the code.